### ISSUE TITLE
mat-chip-grid sometimes adds aria-required=false without no role

### ISSUE THREAD

**Issue by mertdeg2** [angular/components#17397]
#### Bug, feature request, or proposal:
Bug

#### What is the expected behavior?
aria-required should only be used when we have valid role attribute.

#### What is the current behavior?
I am using mdc-chips backed mat-chip-grid implementation. mat-chip-grid component is adding aria-required=false without a role when grid has rows.

#### What are the steps to reproduce?
1. Import MDC chips based mat-chip-grid implementation
2. Paste following template
```
<mat-chip-grid #chipList>
    <mat-chip-row [removable]="true">
      chip
    </mat-chip-row>
    <input [matChipInputFor]="chipList"
                [matChipInputAddOnBlur]="true"/>
  </mat-chip-grid>
```
3. Inspect browser DOM to see mat-chip-grid element has aria-required=false even though it doesn't have role attribute.

#### What is the use-case or motivation for changing an existing behavior?
This is a potential accessibility violation reported by axe. aria-required=false doesn't make sense for non form elements: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-required_attribute

**Comment by mertdeg2** [comment]
Talked to @jelbourn about this bug offline
